lib/prelude.dx

```diff
@@ -406,6 +409,78 @@ def finEq (n:Int) ?-> : Eq (Fin n) = MkEq \x y. ordinal x == ordinal y
 def finOrd (n:Int) ?-> : Ord (Fin n) =
   MkOrd finEq (\x y. ordinal x > ordinal y) (\x y. ordinal x < ordinal y)
 
+'## Raw pointer operations
+
+Int32Ptr : Type = %Int32Ptr
+Word8Ptr : Type = %Word8Ptr
+
+CharPtr = Word8Ptr
+
+data Ptr a:Type = MkPtr Word8Ptr
+
+-- Is there a better way to select the right instance for `storageSize`??
+data TypeVehicle a:Type = MkTypeVehicle
```

Comment:
In Haskell this exact ADT is called `Proxy`, and the usual technique to get around the issue is to use type applications. We wouldn't need it here if this was valid Dex (not sure if it is):
```
def storageSize (a : Type) (_: Storable a) ?=> : Int = ...
```

Reply:
I had assumed that wouldn't work (putting implicit dictionary arguments after explicit arguments) but apparently it does! That's much better. Thanks.

Reply:
If that's valid Dex then why do we still have the `TypeVehicle`?